mobx/lib/src/api/observable_collections/observable_list.dart

```diff
@@ -336,7 +341,7 @@ class ObservableList<T>
       if (iterable.isNotEmpty) {
         final oldValues = _list.sublist(index, index + iterable.length);
         final newValues = iterable.toList(growable: false);
```

Comment:
`newValues` can be created first and then used to fetch the `length`. That way the previous call on `iterable.length` is not needed (which actually forces a complete walk of the iterable)